endpoint.go

```diff
@@ -637,6 +639,14 @@ func EndpointOptionGeneric(generic map[string]interface{}) EndpointOption {
 	}
 }
 
+// CreateOptionIpam function returns an option setter for the ipam configuration for this endpoint
+func CreateOptionIpam(prefAddress net.IP, ipamOptions map[string]string) EndpointOption {
```

Comment:
I was thinking of an `InterfaceIpamConf` structure so that it is easier to add more fields in future. But I cannot think of any other fields, besides the address and options. 
Therefore I think It is better your way, less boilerplate code.